### PR TITLE
[cov] Remove necessity of github connection

### PR DESCRIPTION
**Pull request by steuhs**
Previous PR https://github.com/knative/test-infra/pull/1395 abandoned due to merge conflicts

Addresses comments + rebased

**What this PR does, why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes to reviewers**:

**User-visible changes in this PR**:

### CODE REVIEW

File: tools/coverage/presubmit.go [hunk 1]
```diff
@@ -27,14 +27,19 @@ import (
...
+// RunPresubmit runs the pre-submit procedure
 func RunPresubmit(p *gcs.PreSubmit, arts *artifacts.LocalArtifacts) bool {
 	log.Println("starting PreSubmit.RunPresubmit(...)")
 
-	concernedFiles := githubUtil.GetConcernedFiles(&p.GithubPr, "")
-	if len(concernedFiles) == 0 {
-		log.Printf("List of concerned committed files is empty, " +
-			"don't need to run coverage profile in presubmit\n")
-		return false
+	var concernedFiles map[string]bool
```
Comment (yt3liu) [inline tools/coverage/presubmit.go]: Can you add a comment on what the concerned files for?

File: tools/coverage/calc/delta.go [hunk 2]
```diff
@@ -135,7 +135,7 @@ func (changes *GroupChanges) processChangedFiles(githubFilePaths map[string]bool
 	for i, inc := range changes.Changed {
 		pathFromProfile := githubUtil.FilePathProfileToGithub(inc.base.Name())
 		fmt.Printf("checking if this file is in github change list: %s", pathFromProfile)
-		if githubFilePaths[pathFromProfile] {
+		if len(githubFilePaths) == 0 || githubFilePaths[pathFromProfile] {
```
Comment (yt3liu) [inline tools/coverage/calc/delta.go]: Can you add a comment on why you add the extra check?

**Comment by chaodaiG** [comment]
/approve
/hold

Left a nit, and feel free unhold if you don't want to deal with it now

**Comment by adrcunha** [comment]
> Please update unit test to include the extra branching logic.

+1, please address.

File: tools/coverage/calc/delta.go [hunk 3]
```diff
@@ -132,10 +132,22 @@ func (changes *GroupChanges) processChangedFiles(githubFilePaths map[string]bool
...
+		log.Printf("No github connection. Listing each file with a coverage change.")
+	}
+
 	for i, inc := range changes.Changed {
 		pathFromProfile := githubUtil.FilePathProfileToGithub(inc.base.Name())
-		fmt.Printf("checking if this file is in github change list: %s", pathFromProfile)
-		if githubFilePaths[pathFromProfile] {
+
+		if noRepoConnection {
+			fmt.Printf("File with coverage change: %s", pathFromProfile)
```
[the remark this answers is not in the document]
Comment (steuhs) [reply tools/coverage/calc/delta.go]: I don't think we can use the same message. if noRepoConnection is false we need to do additional check and see if the file appears in commit, whereas if that is true we take all files

**Comment by steuhs** [comment]
> Please update unit test to include the extra branching logic.

Done

File: tools/coverage/presubmit_test.go [hunk 4]
```diff
@@ -33,70 +33,110 @@ const (
...
+	}{
+		{
+			"RunPresubmitWithFakeGithubClient",
+			githubFakes.FakeGithubClient(),
+			false,
+			false,
+		},
+		{
+			"RunPresubmitWithNoGithubClient",
+			githubFakes.FakeGithubClient(),
```
Comment (coryrc) [inline tools/coverage/presubmit_test.go]: Still have it here

Comment (coryrc) [reply tools/coverage/presubmit_test.go]: i.e. I think it should just be an empty string? No github mock?

Comment (steuhs) [reply tools/coverage/presubmit_test.go]: Nice catch. It should be nil

**Pull request merged**